cclient/serialization/json/json_serializer.c

```diff
@@ -106,12 +107,10 @@ static retcode_t flex_hash_array_to_json_array(flex_hash_array_t* const head,
     LL_FOREACH(head, elt) {
       // flex to trytes;
       size_t len_trytes = elt->hash->num_trits / 3;
-      trit_t trytes_out[len_trytes + 1];
-      size_t trits_count =
-          flex_trits_to_trytes(trytes_out, len_trytes, elt->hash->trits,
-                               elt->hash->num_trits, elt->hash->num_trits);
+      char trytes_out[len_trytes + 1];
+      ret = flex_hash_to_trytes(elt->hash, trytes_out);
       trytes_out[len_trytes] = '\0';
-      if (trits_count != 0) {
+      if (ret == RC_OK) {
```

Comment:
Could you adjust the error code below `RC_CCLIENT_FLEX_TRITS` and give it a better name as well?

Reply:
Now, `RC_CCLIENT_FLEX_TRITS` error is returned when trits num is invalid.
How about changing it to `RC_CCLIENT_INVALID_TRITS_LEN`?

And in function `flex_hash_to_trytes`, we will first check if `char* trytes` is null or not, now it returns `RC_CCLIENT_FLEX_TRITS`. But I think `RC_CCLIENT_NULL_PTR` will be more appropriate.